app/controllers/reports_controller.rb

```diff
@@ -92,7 +96,16 @@ def edit
 
   def update
     authorize! :update, @report
-
+    if params[:report][:record_type] == "kpi"
```

Comment:
This if block is very similar to the one above in 'create'.  Extract this common logic into a method, perhaps loaded in a before action.